services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/file/FileHandleManagerImpl.java

```diff
@@ -1040,6 +1041,8 @@ public S3FileHandle createS3FileHandleCopy(UserInfo userInfo, String handleIdToC
 		ValidateArgument.requirement(StringUtils.isNotEmpty(fileName) || StringUtils.isNotEmpty(contentType),
 				"Either the fileName or the contentType needs to be set");
 
+		EntityNameValidation.valdiateName(fileName);
```

Comment:
would it be less confusing to not use the name 'entity validation' if the check also applies to file names, which are not entity names?